pkg/hubble/relay/pool/types/client.go

```diff
@@ -36,3 +46,23 @@ type ClientConnBuilder interface {
 	// optionally, hostname.
 	ClientConn(target, hostname string) (ClientConn, error)
 }
+
+// WaitForReady blocks until the provided connection becomes Ready or timeout is reached.
+func WaitForReady(conn ClientConn, timeout time.Duration) bool {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	for {
+		s := conn.GetState()
+		if s == connectivity.Idle {
+			conn.Connect()
+		}
```

Comment:
You may want to also check if the connection is in a shutdown state and return early if it is.